pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1500,7 +1500,7 @@ func (cf *changeFrontier) maybeProtectTimestamp(
 }
 
 func (cf *changeFrontier) maybeEmitResolved(newResolved hlc.Timestamp) error {
-	if cf.freqEmitResolved == emitNoResolved {
+	if cf.freqEmitResolved == emitNoResolved || newResolved.IsEmpty() {
```

Comment:
When would the empty timestamp actually happen?

Reply:
Empty timestamp would happen when we are doing initial backfill, before we checkpointted high water mark.
What happens then is that we call maybeEmitResovled with empty resolved timestamp -- because that's the frontier when we decide to checkpoint backfilled spans.